yt/frontends/enzo_p/data_structures.py

```diff
@@ -207,12 +207,18 @@ def _parse_index(self):
                 line = buff[bnl:nnl]
                 block_name, block_file = line.split()
 
-                # The B__ block is a negative refinement level
-                # that cannot be used.
                 if "0" not in block_name and "1" not in block_name:
-                    level = -1
-                    left = self.ds.domain_left_edge.d
-                    right = self.ds.domain_right_edge.d
+                    if self.num_grids == 1:
+                        # Because the B__ block is the only block in the grid,
+                        # it has a refinement level of 0
+                        level = 0
+                        rbid = 0
+                    else:
+                        # The B__ block is a negative refinement level
+                        # that cannot be used.
+                        level = -1
+                    left = np.zeros(self.ds.dimensionality)
+                    right = np.ones(self.ds.dimensionality)
```

Comment:
Is there a reason that you're assuming the left edge and right edge arrays here and overriding what you read in from your file? Your fix with the updated logic works if you continue to use `self.ds.domain_left_edge.d` and `self.ds.domain_right_edge.d`.

Reply:
Right, this a subtle change that my toy test don't address. The use of `self.ds.domain_left_edge.d` and `self.ds.domain_right_edge.d` leads to incorrect grid edges. I'll provide a modified example showing this in a following comment (it requires a slightly modified simulation).

Within the for-loop, `left` and `right` effectively have units of "fraction of the domain". After the for-loop, the values of `left` and `right` are re-scaled to have units of `"code_length"` (see lines 276 to 281). Since the highlighted code is handling a single block extending over the full simulation domain, `left=[0,0,0]` and `right=[1,1,1]`. If we assigned values to `left` and `right` based on the results of a call to `get_block_info` (this is done all other blocks on line 223), the values of  `left` and `right` would be the same. If you think it would be more clear, I could replace the highlighted lines with: `left, right = get_block_info(block_name)[1:]`

Reply:
I think I follow this, but I am not sure that memorializing the discussion here will be enough to prevent future confusion.  Could you convert this into a short comment?

Reply:
+1 to adding an inline comment to the code. I think that would be a helpful addition.

Reply:
In the new commit, this should hopefully be unnecessary. Blocks named `B`, `B_`, and `B__` are no longer handled as a special case